src/TreeView/utils.ts

```diff
@@ -37,15 +37,15 @@ export const usePrevious = (x: Set<NodeId>) => {
   return ref.current;
 };
 
-export const usePreviousData = (value: INode[]) => {
-  const ref = useRef<INode[]>();
+export const usePreviousData = <M = IMetadata>(value: INode<M>[]) => {
```

Comment:
This will be just <M> once my other requested changes are made.

Reply:
Could you clarify what you meant here?

Reply:
It was a typo on my part. I was saying that it would be M = unknown after you made my other changes.